core/src/logging/filter/filter_builders.rs

```diff
@@ -0,0 +1,73 @@
+use crate::log_id::LogLevel;
+
+use super::{AddonFilter, FilterConfig, LogIdAddonFilter, LogIdModuleFilter};
+
+#[derive(Default, Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct FilterConfigBuilder {
+    // TODO: negation is confusing, use affirmation instead (no_general_logging -> general_logging)
+    no_general_logging: bool,
+    general_level: LogLevel,
+    general_addons: Vec<AddonFilter>,
+    /// LogIds set with `on[LogId]`
+    allowed_global_ids: Vec<LogIdAddonFilter>,
+    allowed_modules: Vec<LogIdModuleFilter>,
+}
+
+impl FilterConfigBuilder {
+    pub fn new(level: LogLevel) -> Self {
+        Self {
+            general_level: level,
+            ..Default::default()
+        }
+    }
+
+    pub fn level(mut self, level: LogLevel) -> Self {
+        self.general_level = level;
+        self
+    }
+
+    pub fn no_general_logging(mut self) -> Self {
+        self.no_general_logging = true;
+        self
+    }
+
+    pub fn addons<I>(mut self, addons: I) -> Self
```

Comment:
maybe:

```suggestion
    pub fn allow_addons<I>(mut self, addons: I) -> Self
```

To make it clear that added addons are *allowed* by the filter.
Or is it enough to explain it in doc-comments?